### PR TITLE
Limit polling time to 500ms to avoid hanging on shutdown.

### PR DESCRIPTION
**Pull request by impact27**
With spyder, `self.time_to_dead` is 18 seconds. This means that if the code wants to exit, it will have to wait a maximum of 18 seconds while `self.poller.poll(1000 * until_dead)` is processing. This makes smaller polls until the time is reached. Also, it properly processes `ZMQError` on exit.

### CODE REVIEW

File: jupyter_client/channels.py [hunk 1]
```diff
@@ -147,7 +144,10 @@ def run(self):
             # either a recv or connect, which cannot be followed by EFSM
             self.socket.send(b'ping')
             request_time = time.time()
-            ready = self._poll(request_time)
+            ready = False
+            while (not ready and not self._exit.is_set()
+                   and self.time_to_dead > (time.time() - request_time)):
+                ready = self._poll(.5)
```
Comment (rgbkrk) [inline jupyter_client/channels.py]: Is this the only usage of `_poll` in this repo?

Comment (impact27) [reply jupyter_client/channels.py]: Yes

Comment (impact27) [reply jupyter_client/channels.py]: Yes it seems so.

**Comment by SylvainCorlay** [comment]
@impact27 could you please rebase your PR on the top of the latest master?

We reset master to 5.x. The two branches had been diverging too much for no good reason because there was a backward incompatible branch in master that was never really completed.

**Comment by impact27** [comment]
replaced by https://github.com/jupyter/jupyter_client/pull/651

**Pull request closed without merging**